save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestExecutionService.kt

```diff
@@ -82,6 +87,31 @@ class TestExecutionService(private val testExecutionRepository: TestExecutionRep
     internal fun getTestExecutionsCount(executionId: Long) = testExecutionRepository
         .countByExecutionId(executionId)
 
+    /**
+     * @param projectId
+     */
+    internal fun deleteTestExecutionWithProjectId(projectId: Long?) {
+        val testExecutions = testExecutionRepository.findAll { root, cq, cb ->
+            val executionSq: Subquery<Long> = cq.subquery(Long::class.java)
+            val executionRoot: Root<Execution> = executionSq.from(Execution::class.java)
```

Comment:
Vlad, this is not readable, better to use a native query language

Reply:
It won't work. Entity TestExecution don't have link on entity Execution. It have executionId Long

Reply:
Andrey means native SQL query, see [here](https://www.baeldung.com/spring-data-jpa-query).

Reply:
> Andrey means native SQL query, see [here](https://www.baeldung.com/spring-data-jpa-query).

No. It won't work for the same reason.